### ISSUE TITLE
Support for React v15

### ISSUE THREAD

**Issue by simon-lifelike** [mderrick/react-html5video#24]
React v15 [has been released](https://facebook.github.io/react/blog/2016/04/07/react-v15.html), and currently this library has compatibility issues with it. Since `react-addons-pure-render-mixin` is an external dependency and version constrained to `^0.14.0`, which has a peer-dependency of `react@0.14.x`.

Since the mixin is fairly simple, I'd suggest either removing it from `externals` in webpack so it gets bundled with the release (like `lodash.throttle` currently is), or add it as a peer-dependency instead so the responsibility to install a compatible version falls on the user, as with `react` and `react-dom` (why is `react-dom` a peer dependency btw?)

I can submit a pull request, just want to check your thoughts first.

**Comment by mderrick** [comment]
Hi @simon-lifelike thanks for the issue.

I have just made some changes which are up in PR https://github.com/mderrick/react-html5video/pull/25. I've been long thinking about removing this `pure-render-mixin` and now you have given me good reason to do so. I don't think I needed it at all really so instead I have just made sure all the video controls have a `shouldComponentUpdate` method as it is these that would find themselves being rendered extremely often.

I've also gone and removed the dependency on `react-dom` which was just simply a mistake as it needn't be there at all. Let me know if this PR solves your issue.
